es/es.go

```diff
@@ -1,121 +1,87 @@
+		Addresses: []string{url},
+		Username:  username,
+		Password:  password,
+	}
 
-func NewReportNode(params ...string) ReportNode {
-	es := elasticsearch(params...)
-	return ReportNode{es.Client, es.Url, es.username, es.password, "hey-bench", es.Err}
+	client, err := elasticsearch.NewClient(cfg)
+	return Connection{client, url, username, password, err}
 }
 
-func (es ReportNode) FetchReports() ([]commands.TestReport, error) {
-	ret := make([]commands.TestReport, 0)
-
-	search, err := es.Search(es.indexPattern).
-		Sort("@timestamp", false).
-		Size(1000).
-		Do(context.Background())
-	if err != nil {
-		return ret, err
-	}
-
-	for _, hit := range search.Hits.Hits {
-		if err == nil {
-			var report commands.TestReport
-			err = json.Unmarshal(*hit.Source, &report)
-			ret = append(ret, report)
-		}
+func FetchReports(conn Connection, index string) ([]reports.Report, error) {
+	ret := make([]reports.Report, 0)
+	res, err := conn.Search(
+		conn.Search.WithIndex(index),
+		conn.Search.WithSort("@timestamp:desc"),
+	)
+	parsed := SearchResult{}
+	json.NewDecoder(res.Body).Decode(&parsed)
+	for _, hit := range parsed.Hits.Hits {
+		ret = append(ret, hit.Source)
 	}
 	return ret, err
 }
 
-func (es ReportNode) IndexReport(r commands.TestReport) error {
-	_, err := es.Index().
-		Index(es.indexPattern).
-		Type("_doc").
-		BodyJson(r).
-		Do(context.Background())
-	es.Refresh(es.indexPattern).Do(context.Background())
-	return err
-}
-
-type TestNode es
-
-func NewTestNode(params ...string) TestNode {
-	es := elasticsearch(params...)
-	return TestNode{es.Client, es.Url, es.username, es.password, "apm-*", es.Err}
-}
-
-func (es TestNode) Count() int {
-	if es.Err != nil {
-		return -1
+func IndexReport(conn Connection, index string, report reports.Report) error {
+	if conn.Err != nil {
+		return conn.Err
 	}
-	es.Flush(es.indexPattern).Do(context.Background())
-	ret, err := es.Search(es.indexPattern).Do(context.Background())
+	res, err := conn.Index(index, esutil.NewJSONReader(report),
+		conn.Index.WithRefresh("true"),
+		conn.Index.WithDocumentID(report.ReportId),
+	)
 	if err != nil {
-		return -1
-	} else {
-		// exclude onboarding doc
-		return int(math.Max(0, float64(ret.Hits.TotalHits-1)))
+		return err
 	}
-}
+	defer res.Body.Close()
 
-func (es TestNode) Health() (string, error) {
-	if es.Err != nil {
-		return "", es.Err
-	}
-	status, err := es.ClusterHealth().Do(context.Background())
-	if err != nil {
-		return "", err
+	if res.IsError() {
+		return errors.New(res.String())
 	}
-	return strings.TrimSpace(status.Status), nil
+	return nil
 }
 
-// deletes all apm-* indices
-func (es TestNode) Reset() error {
-	if es.Err != nil {
-		return es.Err
+func Count(conn Connection, index string) uint64 {
+	if conn.Err != nil {
+		return 0
 	}
-	_, err := es.Client.DeleteIndex(es.indexPattern).Do(context.Background())
-	if err == nil {
-		_, err = es.Client.Flush(es.indexPattern).Do(context.Background())
+	res, err := conn.Search(
```

Comment:
Hi @jalvz, glad to see that the client is useful! Just a note that for a "count" type of request, it would probably better to use [the `Count()` API](https://github.com/elastic/go-elasticsearch/blob/master/esapi/api.count.go) — that should be a bit more light-weight (doesn't collect hits) and also the `Search()` APIs are currently capped at 10,000 hits in Elasticsearch, if I remember correctly.

Reply:
great, thanks for the tip!